docs/troubleshooting.asciidoc

```diff
@@ -0,0 +1,236 @@
+[float]
+[id="{p}-get-init-container-logs"]
+=== Get init container logs
+
+An Elasticsearch Pod runs a few init containers to prepare the file system of the main Elasticsearch container.
+In some scenarios, the Pod may fail to run (`Status: Error` or `Status: CrashloopBackOff`) because one of the init containers is failing to run.
+Look at the link:https://kubernetes.io/docs/tasks/debug-application-cluster/debug-init-containers/[init container statuses and logs to get more details].
+
+
+[float]
+[id="{p}-get-eck-logs"]
+=== Get ECK logs
+
+Since the ECK operator is just a standard Pod running in the Kubernetes cluster, you can fetch its logs as any other Pod:
+
+[source,sh]
+----
+kubectl -n elastic-system logs -f statefulset.apps/elastic-operator
+----
+
+The operator is constantly attempting to reconcile Kubernetes resources to match the desired state.
+Logs with `INFO` level provide some insights about what is going on.
+
+There should be normally very few logs with `ERROR` level: they indicate something is not going as expected.
+
+Due to link:https://github.com/eBay/Kubernetes/blob/master/docs/devel/api-conventions.md#concurrency-control-and-consistency[optimistic locking],
+there might be some errors reporting a conflict while updating a resource. Those are safe to ignore, since the update should go through at the next reconciliation attempt, which should happen almost immediately.
+
+[float]
+[id="{p}-eck-debug-logs"]
+=== Enable ECK debug logs
+
+To enable `DEBUG` level logs on the operator, restart it with the flag `--enable-debug-logs=true`. For example:
+
+[source,sh]
+----
+kubectl edit statefulset.apps -n elastic-system elastic-operator
+----
+
+and change the following lines from:
+
+[source,yaml]
+----
+  spec:
+    containers:
+    - args:
+      - manager
+      - --operator-roles
+      - all
+      - --enable-debug-logs
+      - false
+----
+
+to
+
+[source,yaml]
+----
+  spec:
+    containers:
+    - args:
+      - manager
+      - --operator-roles
+      - all
+      - --enable-debug-logs
+      - true
+----
+
+[float]
+[id="{p}-pause-controllers"]
+=== Pause ECK controllers
+
+When debugging Elasticsearch, it may be necessary to "pause" the operator reconciliations so no resource gets modified or created in the meantime.
+
+It can be achieved by setting the annotation `common.k8s.elastic.co/pause` to `true` to any resource controlled by the operator :
+
+- Elasticsearch
+- Kibana
+- ApmServer
+
+[source,yaml]
+----
+metadata:
+  annotations:
+    common.k8s.elastic.co/pause: "true"
+----
+
+Or in one line:
+
+[source,sh]
+----
+kubectl annotate elasticsearch quickstart --overwrite common.k8s.elastic.co/pause=true
+----
+
+[float]
+[id="{p}-get-k8s-events"]
+=== Get Kubernetes events
+
+ECK will emit events when:
+
+* important operations are performed (example: a new Elasticsearch Pod was created)
+* something is wrong, and the user should know about it
+
+Fetch Kubernetes events:
+
+[source,sh]
+----
+kubectl get events
```

Comment:
I wonder if it's worth pointing people towards using the `describe` verb here since at least for me that's easier to remember and type than all the field selector stuff

Reply:
Yes maybe. There is a section about `describe` already. I think it's worth pointing out `events` for folks that may ingest k8s events into eg. an Elasticsearch cluster 😄